airbyte-integrations/connectors/source-influxdb/source_influxdb/influxdb.yaml

```diff
@@ -0,0 +1,51 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://ap-southeast-2-1.aws.cloud2.influxdata.com"
+    http_method: "GET"
+    authenticator:
+      type: ApiKeyAuthenticator
+      header: "apikey"
+      api_token: "{{ config['api_key'] }}"
+    request_options_provider:
+      request_parameters:
+        u: "{{ config['u'] }}"
+        p: "{{ config['p'] }}"
+        db: "{{ config['db'] }}"
+        q1:  "{{ config['q1'] }}"
+        q2:  "{{ config['q2'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+
+  wifi_status_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "wifi_status"
+      path: "/query?u={{ config['u'] }}&p={{ config['p'] }}&db={{ config['db'] }}&q={{ config['q1'] }}"
+  
+  airSensors_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "airSensors"
+      path: "/query?u={{ config['u'] }}&p={{ config['p'] }}&db={{ config['db'] }}&q={{ config['q2'] }}"
+  
+streams:
```

Comment:
Hmm, I don' think that you can hardcode streams and stream schemas when it comes to InfluxDB given that users can have different types of data stored on their db. You need to perform some type of schema discovery/exploration in order to show what can be queried or not https://docs.influxdata.com/influxdb/v1.7/query_language/schema_exploration/

Reply:
@itaseskii since influxdb is a time series database, the returned json format is always something like this


    `{
      "results": [
          {
              "statement_id": 0,
              "series": [
                  {
                      "name": "table_name_queried"
                      "columns": [
                        
                      ],
                      "values": [
                        
                      ]
                  }
              ]
          }
      ]
    }`
    
If I have a schema like this, will it work? Please comment.

Reply:
Hey @bala-ceg not really. you schema needs to be [json schema](https://json-schema.org/) compatible https://airbytehq.github.io/understanding-airbyte/supported-data-types/. As for how you want to translate your data model to a valid schema that depends on what operation the database supports, whether it is retrieving table metadata or reading records and inferring the schema from there it is up to you to find out :)

Reply:
Hey @bala-ceg not really. you schema needs to be [json schema](https://json-schema.org/) compatible https://airbytehq.github.io/understanding-airbyte/supported-data-types/. As for how you want to translate your data model to a valid schema that depends on what operation the database supports, whether it is retrieving table metadata or reading records and inferring the schema from there it is up to you to find out :)